File: AD-HOC/2377/main.rb
# frozen_string_literal:true

def total_cost(l, d, k, p)
  distance = l * k
  tolls = (l / d) * p
  distance + tolls
end

l, d = gets.split.map(&:to_i)
k, p = gets.split.map(&:to_i)
puts total_cost(l, d, k, p)

File: AD-HOC/2377/main.c
#include <stdio.h>

int totalCost(int l, int d, int k, int p) {
    int distance = l * k;
    int tolls = (l / d) * p;
    return distance + tolls;
}

int main() {
    int l, d, k, p;
    scanf("%d %d %d %d", &l, &d, &k, &p);
    printf("%d\n", totalCost(l, d, k, p));
    return 0;
}
